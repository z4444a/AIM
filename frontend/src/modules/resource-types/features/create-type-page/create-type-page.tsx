import React from 'react';
import ResourceTypePage from '../resource-type-page/index';
import { mapDispatchToProps } from './index';

export type InternalProps = ReturnType<typeof mapDispatchToProps>;

export class CreateTypePage extends React.PureComponent<InternalProps> {
  public render(): React.ReactNode {
    const { createType } = this.props;
    return <ResourceTypePage onSubmit={createType} />;
  }
}
