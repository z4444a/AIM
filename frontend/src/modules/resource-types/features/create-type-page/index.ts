import { CreateTypePage } from './create-type-page';
import { TypeCreateModel } from '../../../../model/create/type-create-model';
import { createType } from '../../../../redux/actions/type-action';
import { connect } from 'react-redux';

export const mapDispatchToProps = (dispatch: Function) => ({
  createType: (type: TypeCreateModel) => dispatch(createType.request(type)),
});

export default connect(
  null,
  mapDispatchToProps
)(CreateTypePage);
