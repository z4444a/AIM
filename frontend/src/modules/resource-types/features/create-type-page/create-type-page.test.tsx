import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { CreateTypePage, InternalProps } from './create-type-page';

describe('CreateTypePage', () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    const props: InternalProps = {
      createType: () => {},
    };

    wrapper = shallow(<CreateTypePage {...props} />);
  });

  it('should render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
