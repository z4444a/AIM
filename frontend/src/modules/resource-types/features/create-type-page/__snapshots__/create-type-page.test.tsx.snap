// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTypePage should render correctly 1`] = `
<Connect(WithStyles(ResourceTypePage))
  onSubmit={[Function]}
/>
`;
